src/primitives.ts

```diff
@@ -930,180 +930,61 @@ export class Wire {
-    var point_style = PointStyle.deserialize(serialized['point_style']);
-    var surface_style = SurfaceStyle.deserialize(serialized['surface_style'])
-    var edge_style = EdgeStyle.deserialize(serialized['edge_style'])
+    let tooltip = Tooltip.deserialize(serialized['tooltip']);
     return new PieChart(tooltip,
-                       serialized['attribute_names'],
-                       point_style,
-                       surface_style,
-                       edge_style,
-                       serialized['elements'],
-                       axis,
-                       serialized['type_'],
-                       serialized['name']);
-  }
-
-  initialize_all_attributes() {
-    var attribute_names = Object.getOwnPropertyNames(this.elements[0]);
-    var exceptions = ['package_version', 'object_class'];
-    for (let i=0; i<attribute_names.length; i++) {
-      let name = attribute_names[i];
-      if (!List.is_include(name, exceptions)) {
-        let type_ = TypeOf(this.elements[0][name]);
-        this.all_attributes.push(new Attribute(name, type_)); 
-      }
-    }
-  }
-
-  initialize_to_display_attributes() {
-    for (let i=0; i<this.attribute_names.length; i++) {
-      var name = this.attribute_names[i];
-      for (let j=0; j<this.all_attributes.length; j++) {
-        if (name == this.all_attributes[j]['name']) {
-          this.to_display_attributes.push(this.all_attributes[j]);
-          break;
-        }
-      }
-    }
+                        serialized['slicing_variable'],
+                        serialized['data_samples'],
+                        serialized['type_'],
+                        serialized['name']);
   }
 
-  initialize_lists() {
-    this.lists = [];
-    for (let i=0; i<this.to_display_attributes.length; i++) {
-      var value = [];
-      var name = this.to_display_attributes[i].name;
-      let type_ = this.to_display_attributes[i].type_;
-      if (type_ == 'float') {
-        let min = this.elements[0][name];
-        let max = this.elements[0][name];
-        for (let j=0; j<this.elements.length; j++) {
-          let elt = this.elements[j][name];
-          if (elt>max) {
-            max = elt;
-          }
-          if (elt<min) {
-            min = elt;
-          }
-        }
-        this.lists.push([min, max]);
-      } else if (type_ == 'color') {
-        for (let j=0; j<this.elements.length; j++) {
-          let elt_color = rgb_to_string(this.elements[j][name]);
-          if (!List.is_include(elt_color, value)) {
-            value.push(elt_color);
-          }
-        }
-        this.lists.push(value);
-      } else {
-        for (let j=0; j<this.elements.length; j++) {
-          let elt = this.elements[j][name].toString();
-          if (!List.is_include(elt, value)) {
-            value.push(elt);
-          }
-        }
-        this.lists.push(value);
-      }
-    }
-  }
+  computeTotalSample(dataSamples: Object[]): number {
```

Comment:
`computeTotalSample() : number` is enought

Reply:
Done and changed name to computeNormedRatio